mfa/oath/src/main/kotlin/com/pingidentity/mfa/oath/OathCredential.kt

```diff
@@ -0,0 +1,142 @@
+ * @property oathAlgorithm The HMAC algorithm used (SHA1, SHA256, SHA512).
+ * @property digits The number of digits in the generated codes.
+ * @property period For TOTP, the time period in seconds for which a code is valid.
+ * @property counter For HOTP, the counter value used to generate the next code.
+ * @property createdAt The timestamp when this credential was created.
+ * @property imageURL Optional URL for the issuer's logo or image.
+ * @property backgroundColor Optional background color for the credential.
+ * @property policies Optional Authenticator Policies in a JSON String format for the credential.
+ * @property lockingPolicy Optional lName of the Policy locking the credential.
+ * @property isLocked Indicates whether the credential is locked.
+ */
+data class OathCredential(
+    override val id: String = UUID.randomUUID().toString(),
+    override val userId: String = "",
+    override val resourceId: String = "",
+    override val issuer: String,
+    override val displayIssuer: String,
+    override val accountName: String,
+    override val displayAccountName: String,
+    val oathType: OathType,
+    override val secret: String,
+    val oathAlgorithm: OathAlgorithm = OathAlgorithm.SHA1,
+    override val digits: Int = 6,
+    override val period: Int = 30,
+    override val counter: Long = 0L,
+    override val createdAt: Date = Date(),
+    override val imageURL: String? = null,
+    override val backgroundColor: String? = null,
+    override val policies: String? = null,
+    override val lockingPolicy: String? = null,
+    override val isLocked: Boolean = false
+) : MfaOathCredential {
+
+    /**
+     * The type of credential (TOTP or HOTP) as String.
+     */
+    override val type: String
+        get() = oathType.name
+
+    /**
+     * The HMAC algorithm used as String.
+     */
+    override val algorithm: String
+        get() = oathAlgorithm.name
+    
+    companion object {
+        /**
+         * Create an OathCredential from a URI.
+         * Format: otpauth://totp/Issuer:AccountName?secret=SECRET&issuer=Issuer&algorithm=SHA1&digits=6&period=30
+         * Format: otpauth://hotp/Issuer:AccountName?secret=SECRET&issuer=Issuer&algorithm=SHA1&digits=6&counter=0
+         *
+         * @param uri The URI string.
+         * @return An OathCredential.
+         * @throws IllegalArgumentException if the URI is invalid.
+         */
+        @JvmStatic
+        fun fromUri(uri: String): OathCredential {
+            return OathUriParser.parse(uri)
+        }
+    }
+    
+    /**
+     * Convert this credential to a URI string.
+     *
+     * @return A URI string for this credential.
+     */
+    override fun toUri(): String {
+        return OathUriParser.format(this)
+    }
+
+    /**
+     * Converts this credential to a JSON string representation.
+     *
+     * @return A JSON string representing this credential.
+     */
+    override fun toJson(): String {
```

Comment:
Consider using kotlinx.serialization, and use kotlinx.serialization.json intead of JSONObject